OrbitGl/App.cpp

```diff
@@ -329,7 +329,47 @@ void OrbitApp::Disassemble(int32_t pid, const Function& function) {
     disasm.Disassemble(reinterpret_cast<const uint8_t*>(memory.data()),
                        memory.size(), function.GetVirtualAddress(),
                        Capture::GTargetProcess->GetIs64Bit());
-    SendDisassemblyToUi(disasm.GetResult());
+    if (!sampling_report_ || !sampling_report_->GetProfiler()) {
+      SendDisassemblyToUi(disasm.GetResult());
+      return;
+    }
+    std::shared_ptr<SamplingProfiler> profiler =
+        sampling_report_->GetProfiler();
+    unsigned int count_of_function =
+        profiler->GetCountOfFunction(function.GetVirtualAddress());
+    if (count_of_function == 0) {
+      SendDisassemblyToUi(disasm.GetResult());
+      return;
+    }
+
+    const std::function<const double(size_t)> line_to_hits =
+        [profiler, disasm, count_of_function](size_t line) {
```

Comment:
If Disassembler if expensive to copy, you might wanna move it into the lambda: `disasm = std::move(disasm)`. I haven't though if that's the case.

Reply:
Disassembler has no move assignment operator/constructor yet. So without that, the disassembly text becomes empty when moving. 
However, I don't think copying it is to much of an overhead. It basically has a vector<uint64_t> of roughly the size of number instructions and a string with the disassembly.